File: game.Server/Models/Blueprint.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace game.Server.Models
{
    public class Blueprint
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int BlueprintId { get; set; }
        public int ItemId { get; set; }
        public int Price { get; set; }

        [ForeignKey("ItemId")]
        public virtual Item? Item { get; set; }
        public ICollection<Crafting> Craftings { get; set; }

    }
}

File: game.client/src/types/api/models/blueprint.ts
export type Blueprint = {
    blueprintId: number
    price: number
    item: {
        itemId: number
        name: string
        description: string
        itemType: string
        weight: number
        damage: number
        maxDurability: number
    }
    craftings: [
        {
            item: {
                itemId: number
                name: string
            }
            amount: number
        },
    ]
}
